cmd/osbuild-worker/main.go

```diff
@@ -84,6 +90,80 @@ func WatchJob(ctx context.Context, job worker.Job) {
 	}
 }
 
+// protect an AWS instance from scaling and/or terminating.
```

Comment:
This needs a simple test to check whether this doesn't crash if not running on AWS. Just for the sake of sanity, we run a lot of workers outside of AWS.

Reply:
Thanks for the review, in order for it to work an instance needs to be not only on AWS but inside an autoscaling group and to have the access rights to describe its autoscaling group. If these conditions are not fully met, then the process of protecting will silently fail. We keep it silent for the reason that the behavior shall not change at all on non AWS workers. The fact that it passes CI proves that it works outside the required configuration to lock an instance.

And I realize that this bit of text should be in the PR and commit description :sweat_smile:

Reply:
We'd see this in CI already no? Like workers would crash immediately.

Reply:
Yes the fact that this passes CI pretty much shows that it's safe.

Reply:
Fair enough, I still think that a small unit test would be beneficial as it's much simpler to run one and it feels safer but I will leave it up to you.